### PR TITLE
Fixes for Storage and Firestore to conform with the README

### PR DESCRIPTION
**Pull request by jskeet**
(I'll then change my Bigtable PR to conform too.)

### CODE REVIEW

**Comment by jadekler** [comment]
Github makes it quite hard to parse what the heck is going on with 230 files. Could you provide a TLDR?

**Comment by jskeet** [comment]
Whoops, absolutely:

Commit 1:

- Move storage/v1/test.proto to storage/v1/proto/google/cloud/conformance/storage/v1/tests.proto
- Change validator accordingly

Commit 2:

- Remove firestore/v1/README (now redundant)
- Move files from firestore/v1/testcase to firestore/v1
- Change validator accordingly

**Comment by jadekler** [comment]
Thank you! That all SGTM and seems in line with the specification you wrote.
